cmake/LLVM.cmake

```diff
@@ -392,16 +392,20 @@ find_path(LLVM_SPIRV_INCLUDEDIR "LLVMSPIRVLib.h"
 find_library(LLVM_SPIRV_LIB "LLVMSPIRVLib" PATHS "${LLVM_LIBDIR}" NO_DEFAULT_PATH)
 
 # Ubuntu's libllvmspirv-XY-dev packages unfortunately use unversioned
-# /usr/include and /usr/lib/<arch>/libLLVMSPIRVLib.so
-find_path(LLVM_SPIRV_INCLUDEDIR "LLVMSPIRVLib.h" PATH_SUFFIXES "LLVMSPIRVLib")
-find_library(LLVM_SPIRV_LIB "LLVMSPIRVLib" PATHS "${LLVM_LIBDIR}")
-
-if(LLVM_SPIRV_INCLUDEDIR AND LLVM_SPIRV_LIB)
-  message(STATUS "found LLVMSPIRV library: ${LLVM_SPIRV_INCLUDEDIR} | ${LLVM_SPIRV_LIB}")
-  set(HAVE_LLVM_SPIRV_LIB 1)
-else()
-  message(STATUS "LLVMSPIRV library not found: ${LLVM_SPIRV_INCLUDEDIR} | ${LLVM_SPIRV_LIB}")
-  set(HAVE_LLVM_SPIRV_LIB 0)
+# /usr/include and /usr/lib/<arch>/libLLVMSPIRVLib.so. Because of this
+# we prefer llvm-spirv if the user explicitly requested it by setting
+# the LLVM_SPIRV variable.
+set(HAVE_LLVM_SPIRV_LIB 0)
+if (NOT DEFINED LLVM_SPIRV)
```

Comment:
The problem i have with this, is that this variable (`LLVM_SPIRV`) is set by `find_program()` in  `cmake/LLVM.cmake`, so your comment:

"preferring llvm-spirv tool if the user sets -DLLVM_SPIRV in the configuration"

... is not quite correct; `if(DEFINED ..)` does not distinguish between user-set variables and cache variables. So this code will result in *always* preferring llvm-spirv tool if it's found, even if a library is also available. IMO a better default order would be:

 1) versioned library
 2) if not found, versioned tool
 3) if not found, unversioned library / tool

Reply:
Fixed. It now does as explained in the comment.